sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_connection.py

```diff
@@ -90,6 +90,9 @@ def __init__(self, endpoint, **kwargs):
         # type(str, Any) -> None
         parsed_url = urlparse(endpoint)
         self._hostname = parsed_url.hostname
+        custom_parsed_url = urlparse(kwargs.get("custom_endpoint_address"))
```

Comment:
kwargs.get("") can return None, which I think would raise an error if passed into urlparse. So these operations should probably be separated.